src/model/channel/attachment_type.rs

```diff
@@ -19,75 +19,95 @@ use crate::error::{Error, Result};
 #[non_exhaustive]
 pub enum AttachmentType<'a> {
     /// Indicates that the [`AttachmentType`] is a byte slice with a filename.
-    Bytes { data: Cow<'a, [u8]>, filename: String },
+    Bytes { data: Cow<'static, [u8]>, filename: Cow<'static, str> },
     /// Indicates that the [`AttachmentType`] is a [`File`]
-    File { file: &'a File, filename: String },
+    File { file: &'a File, filename: Cow<'static, str> },
     /// Indicates that the [`AttachmentType`] is a [`Path`]
     Path(&'a Path),
     /// Indicates that the [`AttachmentType`] is an image URL.
     Image(Url),
 }
 
+async fn data_path(path: &Path) -> Result<Vec<u8>> {
```

Comment:
```suggestion
#[cfg(feature = "http")]
async fn data_path(path: &Path) -> Result<Vec<u8>> {
```

You should feature-gate these, as they're only used if `http` is enabled (would also fix the import error).